py4DSTEM/process/diffraction/crystal.py

```diff
@@ -85,22 +85,32 @@ def from_pymatgen_structure(structure, conventional_standard_structure=True):
         """
         import pymatgen as mg
         from pymatgen.ext.matproj import MPRester
-
-        if isinstance(structure, str):
-            with MPRester() as m:
-                structure = m.get_structure_by_material_id(structure)
-
-        assert isinstance(
-            structure, mg.core.Structure
-        ), "structure must be pymatgen Structure object"
-
-        structure = (
-            mg.symmetry.analyzer.SpacegroupAnalyzer(
-                structure
-            ).get_conventional_standard_structure()
-            if conventional_standard_structure
-            else structure
-        )
+        
+        if structure is not None:
+            if isinstance(structure, str):
+                mpr = MPRester(MaPKey)
+                structure = mpr.get_structure_by_material_id(structure)
+            
+            assert isinstance(
+                structure, mg.core.Structure
+            ), "structure must be pymatgen Structure object"
+
+            structure = (
+                mg.symmetry.analyzer.SpacegroupAnalyzer(
+                    structure
+                ).get_conventional_standard_structure()
+                if conventional_standard_structure
+                else structure
+            )
+        else:
+            mpr = MPRester(MaPKey)
+            if formula is None:
+                print('Please provide a formula to query MP databsae')
+            query = mpr.query(criteria={"pretty_formula": formula}, properties=["structure","icsd_ids","spacegroup"])
+            if space_grp:
+                query = [query[i] for i in range(len(query)) if mg.symmetry.analyzer.SpacegroupAnalyzer(query[i]['structure']).get_space_group_number() == space_grp]
+            selected = query[np.argmin([query[i]['structure'].lattice.volume for i in range(len(query))])]
+            structure = mg.symmetry.analyzer.SpacegroupAnalyzer(selected["structure"]).get_conventional_standard_structure()
```

Comment:
This does not seem to obey the user's specified `conventional_standard_structure` setting

Reply:
can you explain this one?

Reply:
The function takes an argument `conventional_standard_structure`, that sets whether to use the conventional standard cell or the primitive cell that pymatgen returns. This line is using the conventional standard structure regardless of what the user specified in the arguments.

Reply:
Got it